src/sql/src/parser.rs

```diff
@@ -187,8 +187,13 @@ impl<'a> ParserContext<'a> {
     }
 
     pub fn consume_token(&mut self, expected: &str) -> bool {
-        if self.peek_token_as_string().to_uppercase() == *expected.to_uppercase() {
-            let _ = self.parser.next_token();
+        Self::inner_consume_token(&mut self.parser, expected)
+    }
+
+    #[inline]
+    pub(crate) fn inner_consume_token(parser: &mut Parser<'a>, expected: &str) -> bool {
+        if Self::inner_peek_token_as_string(parser).to_uppercase() == *expected.to_uppercase() {
```

Comment:
We can call `eq_ignore_ascii_case` instead of allocating two new string using `to_uppercase()`
https://doc.rust-lang.org/stable/std/string/struct.String.html#method.eq_ignore_ascii_case